frontend/src/metabase/modes/components/drill/ColumnFilterDrill.tsx

```diff
@@ -3,26 +3,36 @@ import React from "react";
 import { t } from "ttag";
 import { TYPE, isa } from "metabase/lib/types";
 
+import {
+  ClickAction,
+  ClickActionProps,
+} from "metabase-types/types/Visualization";
+
 import FilterPopover from "metabase/query_builder/components/filters/FilterPopover";
 
 const INVALID_TYPES = [TYPE.Structured];
 
-export default function ColumnFilterDrill({ question, clicked }) {
+export default function ColumnFilterDrill({
+  question,
+  clicked,
+}: ClickActionProps): ClickAction[] {
   const query = question.query();
   if (
     !question.isStructured() ||
     !query.isEditable() ||
     !clicked ||
     !clicked.column ||
-    INVALID_TYPES.some(type => isa(clicked.column.base_type, type)) ||
+    INVALID_TYPES.some(
+      type => clicked.column?.base_type && isa(clicked.column.base_type, type),
+    ) ||
     clicked.column.field_ref == null ||
     clicked.value !== undefined
   ) {
     return [];
   }
 
   const { dimension } = clicked;
-  const initialFilter = dimension.defaultFilterForDimension();
+  const initialFilter = dimension?.defaultFilterForDimension();
```

Comment:
Should we offer this drill at all if a dimension is missing?

Reply:
There's a web of assumptions in the drill through components that started clashing when I added annotations. 

In this case it shouldn't happen but when the filter popover gets an `undefined` initial filter it shows the field picker. I could use an assertion to make sure that `dimension` is not undefined and throw, but that'll go through the React error boundary and show an error, which I think is worse from a UX perspective.

Reply:
Should we maybe just return `[]` then not to offer the filter at all?